### ISSUE TITLE
Callable compared with is True not flagged

### ISSUE THREAD

**Issue by acampove** [astral-sh/ruff#25145]
### Description

When a method or function is used in a conditional with `is True` (instead of being called), ruff does not report any warning or error. Since a callable can never be identical to the `True` singleton, this comparison is always `False`, making the branch dead code.

This is a type-level fact that ruff has all the information to detect: if the left-hand side of an `is True` expression is typed as a callable, the result is always `False`.

### Reproduction

```python
"""Module used to test Ruff."""

import logging

logging.basicConfig(level=logging.INFO)
logger = logging.getLogger(__name__)


class Sample:
    """Test class."""

    def is_merged(self) -> bool:
        """Return true always."""
        return True

sample = Sample()

if sample.is_merged is True:
    logger.info("merged")
```

### Expected behavior

ruff should report a warning/error such as:

> Comparison of a callable with `True` using `is` will always be `False`. Did you mean `sample.is_merged()`?

### Actual behavior

```
warning: `incorrect-blank-line-before-class` (D203) and `no-blank-line-before-class` (D211) are incompatible. Ignoring `incorrect-blank-line-before-class`.
warning: `multi-line-summary-first-line` (D212) and `multi-line-summary-second-line` (D213) are incompatible. Ignoring `multi-line-summary-second-line`.
All checks passed!
```

### Environment

- ruff version: 0.15.12
- Python version: 3.13

**Comment by ntBre** [comment]
I think this would be a better fit for a type checker like ty. I don't think there's an existing lint rule in ty to emit a diagnostic for this, but I can see in the [playground](https://play.ty.dev/a477abdb-9af3-4839-bade-12959cde6c44) that it knows the code is unreachable. It looks like we're tracking such a lint in https://github.com/astral-sh/ty/issues/1948.

As noted on that issue, [mypy](https://mypy-play.net/?mypy=latest&python=3.12&flags=warn-unreachable&gist=7a3cb5def3a60fd4848cf46102c7c39c) also catches this with the `--warn-unreachable` flag enabled.

**Comment by acampove** [comment]
@ntBre Interesting, the pyright maintainers told me that this is a linting issue rather than a type analizing issue:

https://github.com/microsoft/pyright/issues/11442

I cannot move to tz, it is not mature yet, there are many feature it has not implemented, maybe in a year or two it would make more sense to try it.

**Comment by ntBre** [comment]
That is interesting! But I'd say it makes sense. This is kind of on the boundary between a linter and type checker. For us, it definitely makes more sense to implement this in ty because ty has multi-file analysis (Ruff could potentially catch the example you included, but not if `Sample` were defined in another file) and because it already has good handling for reachability analysis.

And ty is getting to be pretty mature! If there are specific issues you've encountered that aren't already tracked in its issue tracker, we'd be glad to know!

For now I'll close this as a duplicate of https://github.com/astral-sh/ty/issues/1948 since that seems like the most direct route forward for one of our tools.
